### DISCUSSION TITLE
Manifest error: "The file name doesn't exist in the package".

### DISCUSSION THREAD

**Discussion by snivets** [microsoft/microsoft-ui-xaml#8398] [Q&A]
Asked on Stack Overflow but didn't get an answer - I'm trying to publish a packaged app to the Windows Store. Using the pack and publish option, I get a failure with this info: "The file name "AudioChapterEditor.exe" declared for element "[local-name()='Applications']/[local-name()='Application']" doesn't exist in the package."

My app is called Audio Chapter Editor, and the solution is titled AudioChapterEditor. The appxmanifest is below, where I assume it's having the problem. (Minor edits for privacy maybe? Not sure what's private in this file!)

The rescap line at the bottom is highlighted in Visual Studio with a warning about being unknown, but it's what the documentation says should be in a WinUI 3 project, and I think it's what was created when I made the project in VS... it seems this kind of error comes up for others when bundling things, particularly executables, into the application, but my app doesn't need or have that.

```
<Package
  xmlns="http://schemas.microsoft.com/appx/manifest/foundation/windows10"
  xmlns:mp="http://schemas.microsoft.com/appx/2014/phone/manifest"
  xmlns:uap="http://schemas.microsoft.com/appx/manifest/uap/windows10"
  xmlns:rescap="http://schemas.microsoft.com/appx/manifest/foundation/windows10/restrictedcapabilities"
  IgnorableNamespaces="uap rescap">

  <Identity
    Name="xxxxxNatex.AudioChapterEditor"
    Publisher="CN=XXXXXXXX-F522-4723-8422-EB217E3E071F"
    Version="1.0.2.0" />

  <mp:PhoneIdentity PhoneProductId="XXXXXXXX-39f8-4588-a25d-d72b56c6b2c0" PhonePublisherId="00000000-0000-0000-0000-000000000000"/>

  <Properties>
    <DisplayName>Audio Chapter Editor</DisplayName>
    <PublisherDisplayName>Nate x</PublisherDisplayName>
    <Logo>Assets\Square150x150Logo.png</Logo>
  </Properties>

  <Dependencies>
    <TargetDeviceFamily Name="Windows.Universal" MinVersion="10.0.17763.0" MaxVersionTested="10.0.19041.0" />
    <TargetDeviceFamily Name="Windows.Desktop" MinVersion="10.0.17763.0" MaxVersionTested="10.0.19041.0" />
  </Dependencies>

  <Resources>
    <Resource Language="x-generate"/>
  </Resources>

  <Applications>
    <Application Id="App"
      Executable="$targetnametoken$.exe"
      EntryPoint="$targetentrypoint$">
      <uap:VisualElements
        DisplayName="Audio Chapter Editor"
        Description="A tool that enables easy and stylish editing of audio chapter files in Windows 11. Designed for podcasters."
        BackgroundColor="transparent"
        Square150x150Logo="Assets\Square150x150Logo.png"
        Square44x44Logo="Resources\BarCafeChair.png">
        <uap:DefaultTile Wide310x150Logo="Assets\Wide310x150Logo.png" />
      </uap:VisualElements>
    </Application>
  </Applications>

  <Capabilities>
    <rescap:Capability Name="runFullTrust" />
  </Capabilities>
</Package>
```

**Comment by snivets** [comment]
Update, this is only happening when I try to build and package for X86. If I don't do that, there's no problems.

**Reply by ErikSJan** [reply]
Hello.

I have exactly the same problem but with x64 instead of x86.
The problem occurs both with VS 2022 17.8.0 and 2022 Preview17.9.0 prev 1 .
It's very easy to reproduce: 
- create a new MAUI application
- create a solution configuration Release | x64
- create a new publish profile the above config, with net8.0-windows10.0.19041.0, win-x64, autonomous, onefile, readytorun
- click on publish and you will get the error

2>C:\Program Files (x86)\Microsoft Visual Studio\Shared\NuGetPackages\microsoft.windows.sdk.buildtools\10.0.22621.756\bin\10.0.22621.0\x64\MakeAppx.exe pack /l /h sha256 /f "obj\x64\Release pour Windows\net8.0-windows10.0.19041.0\win-x64\package.map.txt" /o /p "bin\x64\Release pour Windows\net8.0-windows10.0.19041.0\win-x64\AppPackages\NusaDuaManager_1.0.0.1_Release pour Windows_Test\NusaDuaManager_1.0.0.1_x64_Release pour Windows.msix"  
2>Manifest validation error: Line 26, Column 27, Reason: The file name "NusaDuaManager.exe" declared for element "*[local-name()='Applications']/*[local-name()='Application']" doesn't exist in the package.
2>Package creation failed.
2>0x80080204 - The specified package format is not valid: The package manifest is not valid.

**Comment by DarranRowe** [comment]
First of all, the runFullTrust capability is [properly documented](https://learn.microsoft.com/en-us/windows/uwp/packaging/app-capability-declarations). It is the "Full Trust Permission Level" capability in the restricted capability list.
What's more, both the runFullTrust capability and the [Application declaration documentation](https://learn.microsoft.com/en-us/uwp/schemas/appxpackage/uapmanifestschema/element-application) both state that if your application is full trust then it requires this capability. The documentation has this associated to the mediumIl trust level, but this is something that was added with Windows 10 2004/20H1. The equivalent functionality for versions prior to this is to set the entry point to windows.fullTrustApplication. The reason this is important to point out is the manifest tooling in Visual Studio defaults to just adding the entry point to the manifest, it defaults to full trust and windows.fullTrustApplication is equivalent to declaring the runtime behaviour as packagedClassicApp and the trust level as mediumIl for the 2004/20H1 declarations.

In other words, the warning can be ignored and if you have somehow set the package to be partial trust then you can also remove this capability.

For your actual problem, this is difficult since I honestly can't replicate this. The error I get when the build fails or there is some other problem is completely different from the one in your post. If possible, could you create a sample project or copy your project and cut out anything private to provide a sample? If it is in the packaging portion, you don't have to provide any of the code or resources for the application.

**Reply by snivets** [reply]
I appreciate the clarification there - far as I understand it, WinUI 3 apps require runFullTrust capability, therefore my app is a-ok to generate this warning. I'm new to Windows dev so I'm still learning all the differences between sandboxed and not, WinUI and UWP.

As for replicating, I'm not exactly sure if anything here should be private (let me know if so!) but here's the GitHub repository in question (still currently having that problem): https://github.com/snivets/AudioChapterEditor

**Reply by DarranRowe** [reply]
Well, I tried your project, and packaging up sideloading packages on Visual Studio 2022 17.5 works. I get no errors at all, not even for x86.

So this means that it is either the act of publishing it to the store that is the problem, or there is an issue in the version of Visual Studio that you are using. At the very least, Visual Studio has no issues packing it up and then Windows has no issues installing and running it.

**Reply by snivets** [reply]
That's quite strange, but very helpful to hear. "Works on my machine!" Thank you for looking at this. I might pull the project up on another machine of mine if I need an x86 build.
